### ISSUE TITLE
Stuck on replacing controlplane nodes

### ISSUE THREAD

**Issue by farodin91** [kubernetes-sigs/cluster-api#3127]
**What steps did you take and what happened:**

- Create a Cluster
- Wait for the cluster to be running
- Change the configuration of a controlplane
- New controlplane node are spawned
- New controlplane stucks in the phase: Provisioning with the event: `Waiting for control plane to pass control plane health check to continue reconciliation: control plane machine namespace/node-cp-v2-s8p8w has no status.nodeRef`

**What did you expect to happen:**
* new controlplane node gets the nodeRef and switches to the phase running

**Anything else you would like to add:**
logs of capi-controller-manager 
```
E0529 12:06:21.142555       1 machine_controller_noderef.go:98] controllers/Machine "msg"="Failed to parse ProviderID" "error"="providerID is empty" "providerID"={} "node"="node-cp-v2-s8p8w"
```

I would like to help. I already opened the issue on provider side. I wasn't sure were to open it. https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/issues/921.

**Environment:**

- Cluster-api version: v0.3.6
- Cluster-api-provider-vsphere version: v0.6.4
- Kubernetes version: (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.0", GitCommit:"9e991415386e4cf155a24b1da15becaa390438d8", GitTreeState:"clean", BuildDate:"2020-03-25T14:58:59Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.3", GitCommit:"06ad960bfd03b39c8310aaf92d1e7c12ce618213", GitTreeState:"clean", BuildDate:"2020-02-11T18:07:13Z", GoVersion:"go1.13.6", Compiler:"gc", Platform:"linux/amd64"}
```
- OS (e.g. from `/etc/os-release`): ubuntu 18.04

/kind bug
[One or more /area label. See https://github.com/kubernetes-sigs/cluster-api/labels?q=area for the list of labels]

**Comment by farodin91** [comment]
I checkout the issue a bit deeper and seems the cloud controller in the provision cluster is unable to set the providerID. I think it would helpful to improve the log message.

**Comment by vincepri** [comment]
Makes sense to improve the error message, +1

/milestone v0.3.x

**Comment by vincepri** [comment]
Let's switch to a condition / reason message for Ready to present this information to users, this will be fixed in #3138 

/close
